### PR TITLE
Feat/enable wallet connect by default

### LINKED ISSUE #15673: [dApps] enable dApps feature flag by default

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closes #15673

This PR enables the wallet connect feature by default. The feature flag remains in place.

It brings in new user stories because the WC and BC overlaps in the UI. Therefore the qml components need to work with either of these features disabled.

#### Changes
1. Enable WC by default
2. Account for the WC and BC feature flags in the dapp list providers
3. Account for the WC and BC feature flags in the `Connect a dApp` modal
4. Add qml test

### New user stories identified in the process:

```
Given the Wallet Connect feature is disabled
And Browser Connect feature is enabled
Then the dApps button is visible in the wallet header
```

```
Given the Wallet Connect feature is disabled
And Browser Connect feature is enabled
When `Connect a dApp` button is clicked
Then the `Connect a dApp` modal is shown
```

```
Given the Wallet Connect feature is disabled
And Browser Connect feature is enabled
When the `Connect a dApp` modal is shown
The Wallet Connect button is disabled
```

```
Given the Wallet Connect feature is enabled
And Browser Connect feature is disabled
When the `Connect a dApp` modal is shown
The Browser Connect button is disabled
```

```
Given the Wallet Connect feature is enabled
And Browser Connect feature is disabled
When the `dApps` button is clicked
Then the popup doesn't show BC connected dapps
```

```
Given the Wallet Connect feature is disabled
And Browser Connect feature is enabled
When the `dApps` button is clicked
Then the popup doesn't show WC connected dapps
```

### Affected areas

Wallet Connect
Browser Connect

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/user-attachments/assets/712392c8-df54-4b82-bace-150000c9e7ae

### CODE REVIEW

**Comment by alexjba** [comment]
@benjthayer Please let me know what you think of the behaviour described in the User stories above. In a nutshell, when the Wallet Connect feature is enabled and the Browser Connect is disabled we'll still show the `Connect a dApp` modal as the entry point when connecting dapps, but the feature disabled in the feature flag is disabled in the modal. I think it's nice to show the option as a preview even if it's not ready for the release.
This behaviour is depicted in the screen capture above.

**Review by caybro — APPROVED** [review]
LGTM

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml [hunk 1]
```diff
@@ -18,7 +18,12 @@ QObject {
...
     readonly property var dappsModel: d.dappsModel
 
+    property bool enabled: true
+
     Component.onCompleted: {
+        if (!enabled) {
+            return
+        }
         // Just in case the SDK is already initialized
         d.updateDappsModel()
```
Comment (micieslak) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml:28]: shouldn't it be done on enabled changed to true?

**Review by micieslak — APPROVED** [review]
LGTM, one small question onyl

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml (hunk 1 above)
Comment (alexjba) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml:28]: yes, that's true! Thank you 🍻 
Will update

**Pull request merged**